src/msg.jl

```diff
@@ -34,15 +34,18 @@ include("kwdef.jl")
 function _message(classname, perf, sdef)
   if @capture(sdef, struct T_ <: P_ fields__ end)
     if T == P
+      extra2 = :( Fjage.classname(::Type{$(T)}) = $(classname) )
       T = Symbol("_" * string(T))
```

Comment:
Maybe the modified name could be `T = Symbol("_Concrete" * string(T))` (e.g. `_ConcreteDatagramReq`)? Doesn't cost us anything, and maybe the `Concrete` gives a user who discovers the concrete types in a somewhat haphazard way enough of a hint so they can figure out what's going on (and even if not, `_Concrete` is much easier to search for than just `_`).

Reply:
No objection. But then do we even want the `_` or can we just go with `Concrete` prefix and assume that it is unlikely to conflict with message names? Open either way, for your opinion...

Reply:
I'd say no harm to keep the `_`? It doesn't bother us in any way, it further decreases the risk of name conflicts, and if nothing else it conveys that the concrete types are meant to be internal-only.